### ISSUE TITLE
AuthService and logic relating to login needs reviewing

### ISSUE THREAD

**Issue by JamesMilnerUK** [3drepo/3drepo.io#771]
There's a lot going on in AuthService and HomeController that's quite hard to follow and could probably be simplified quite considerably. I think some refactoring would be easier for others to understand and work with and hopefully make the code less brittle.

Really there are only two states (logged in and logged out) and the only edge case is we need to handle login changes across tabs.

**Comment by carmenfan** [comment]
going to be obsolete due to react migration
